src/main/java/org/openstreetmap/atlas/geography/atlas/change/ChangeEdge.java

```diff
@@ -28,6 +29,18 @@
     private final Edge source;
     private final Edge override;
 
+    // Computing Parent Relations is very expensive, so we cache it here.
+    private transient Set<Relation> relationsCache;
+    private transient Object relationsCacheLock = new Object();
+
+    // Computing Start Node is very expensive, so we cache it here.
```

Comment:
Why was computing the start node expensive? Wouldn't we just check the parent `ChangeAtlas` for the presence of a source/override, and return the appropriate one? There must be an extra step there I am glossing over.

Reply:
Yes, as if a start node is not present, we apply the forgiveness principle and we remove that edge, which in turn removes it from the other connected node, and from the parent relation etc.

Reply:
Ah right forgot about that implication. In that case this makes sense.